File: 2418-sort-the-people/2418-sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        arr=[]
        for i in range(len(names)):
            arr.append([heights[i], names[i]])
        return [i[1] for i in sorted(arr, reverse=True)]
        # arr.sort(reverse=True)
        # return [i[1] for i in arr]
        

File: 2418-sort-the-people/2418-sort-the-people.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        var map = new TreeMap<Integer, String>(Collections.reverseOrder());
        for(int i=0; i<names.length; ++i) map.put(heights[i], names[i]);
        
        int i=0;
        for(int h: map.keySet()){
            names[i] = map.get(h);
            ++i;
        }
        
        return names;
    }
}